crafting-dead-core/src/main/java/com/craftingdead/core/capability/gun/DefaultGun.java

```diff
@@ -672,6 +681,9 @@ public void toggleRightMouseAction(ILiving<?, ?> living, boolean sendUpdate) {
     if (!this.performingRightMouseAction && living.getEntity().isSprinting()) {
       return;
     }
+    if(this instanceof AimableGun && !((AimableGun) this).canAim()){

```

Comment:
No need for this instance check. Override `toggleRightMouseAction` in `AimableGun` and call the super method if `canAim` is true.